### PR TITLE
Available frameworks refactoring

### PR DESCRIPTION
**Pull request by ZiadAmerr**

### CODE REVIEW

**Comment by ZiadAmerr** [comment]
This PR is still a draft until we fix the local tests problem.

File: ivy_tests/test_ivy/helpers/available_frameworks.py [hunk 1]
```diff
@@ -2,57 +2,21 @@
...
 
 def available_frameworks():
-    available_frameworks_lis = ["numpy", "jax", "tensorflow", "torch", "paddle"]
-    try:
-        import jax
+    frameworks = ["numpy", "jax", "tensorflow", "torch", "paddle"]
+    for fw in frameworks:
+        try:
+            exec(f"import {fw}")
+            assert exec(fw), f"{fw} is imported to see if the user has it installed"
```
Comment (ZiadAmerr) [inline ivy_tests/test_ivy/helpers/available_frameworks.py:9]:
```suggestion
```

File: ivy_tests/test_ivy/helpers/available_frameworks.py [hunk 2]
```diff
@@ -2,57 +2,20 @@
...
-        g_truth = "mxnet"
-    else:
-        g_truth = "numpy"
-    return g_truth
+    frameworks = available_frameworks()
+    gt_frameworks = ["tensorflow", "torch", "jax", "paddle", "mxnet"]
+    return next((gt_framework
+                 for gt_framework in gt_frameworks if
+                 gt_framework in frameworks),
+                "numpy")
```
Comment (ZiadAmerr) [inline ivy_tests/test_ivy/helpers/available_frameworks.py:21]:
```suggestion
                "numpy")

```
Added eof line

**Comment by CatB1t** [comment]
Hello @ZiadAmerr, this have been fixed in the integration of `with_backend` and `mp` branch.

**Pull request closed without merging**